main/src/ca/uwaterloo/flix/language/phase/Monomorph.scala

```diff
@@ -154,6 +154,8 @@ object Monomorph extends Phase[TypedAst.Root, TypedAst.Root] {
           val newSym = specializeDefSym(sym, subst0(tpe))
           Expression.Def(newSym, subst0(tpe), loc)
 
+        case Expression.Sig(sym, tpe, loc) => Expression.Sig(sym, subst0(tpe), loc) // MATT surely this should match Def?
```

Comment:
So, eventually, a lot of magi will need to happen here.

This is where we will need to find an appropriate type class *instance* (something we have not yet dealt with) and instantiate it. 

For now, how about you replace it with an expression hole or some kind of error expression.